File: Backend/Movieverse.Contracts/DataTransferObjects/Media/MediaDto.cs
using Movieverse.Domain.ValueObjects;

namespace Movieverse.Contracts.DataTransferObjects.Media;

public class MediaDto
{
	public string Title { get; set; } = null!;
	public Details Details { get; set; } = null!;
	public TechnicalSpecs TechnicalSpecs { get; set; } = null!;
	public int? CurrentPosition { get; set; }
	public BasicStatistics BasicStatistics { get; set; } = null!;
	public Guid? PosterId { get; set; }
	public Guid? TrailerId { get; set; }
}

File: Frontend/movieverse/src/core/dtos/media/mediaDto.ts
import { BasicStatisticsDto } from "./basicStatisticsDto"
import { DetailsDto } from "./detailsDto"
import { StaffDto } from "./staffDto"
import { TechnicalSpecsDto } from "./technicalSpecsDto"

export interface MediaDto {
  title: string,
  details: DetailsDto,
  technicalSpecs: TechnicalSpecsDto,
  currentPosition: number | null,
  basicStatistics: BasicStatisticsDto,
  posterId: string | null,
  trailerId: string | null,
  platformIds: string[],
  genreIds: string[],
  staff: StaffDto[]
};

export interface MovieDto extends MediaDto {
  sequelId: string | null,
  sequelTitle: string | null,
  prequelId: string | null,
  prequelTitle: string | null
};

export interface SeriesDto extends MediaDto {
  seasonCount: number,
  episodeCount: number,
  seriesEnded: string | null
};